docs/grammar/SolidityParser.g4

```diff
@@ -179,10 +179,18 @@ locals[
 /**
  * The definition of a free function.
  */
- freeFunctionDefinition:
+ freeFunctionDefinition
+ locals[
+ 	boolean mutabilitySet = false,
+ 	boolean suffixSpecifierSet = false,
+ ]
+ :
  	Function (identifier | Fallback | Receive)
  	LParen (arguments=parameterList)? RParen
- 	stateMutability?
+ 	(
+		{!$mutabilitySet}? stateMutability {$mutabilitySet = true;}
+		| {!$suffixSpecifierSet}? Suffix {$suffixSpecifierSet = true;}
+ 	 )*
```

Comment:
With just mutability and suffix we could still do it without variables and keep it simple:
```suggestion
	stateMutability Suffix?
	| Suffix stateMutability?
```